File: src/main/java/com/sce/data/gaia/core/CustomGrantedAuthority.java
package com.sce.data.gaia.core;

import org.springframework.security.core.GrantedAuthority;

/**
 * @author bk201
 */
public class CustomGrantedAuthority implements GrantedAuthority {

    private String authority;

    public CustomGrantedAuthority(String authority) {
        this.authority = authority;
    }


    @Override
    public String getAuthority() {
        return null;
    }
}

File: src/main/java/com/sce/data/gaia/core/CustomGrantedAuthority.kt
package com.sce.data.gaia.core

import org.springframework.security.core.GrantedAuthority

/**
 * @author bk201
 */
class CustomGrantedAuthority(private val authority: String) : GrantedAuthority {


    override fun getAuthority(): String {
        return authority
    }
}
